modules/apps/29-fee/keeper/keeper.go

```diff
@@ -85,3 +77,22 @@ func (k Keeper) ClaimCapability(ctx sdk.Context, cap *capabilitytypes.Capability
 	return k.scopedKeeper.ClaimCapability(ctx, cap, name)
 }
 */
+
+// SetCounterPartyAddress maps the counterparty relayer address to the source relayer address
+func (k Keeper) SetCounterpartyAddress(ctx sdk.Context, SourceAddress, CounterpartyAddress string) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeySourceAddress(SourceAddress), []byte(CounterpartyAddress))
```

Comment:
This mapping is incorrect. We are mapping the address on the destination chain -> address on the source chain. 
```
It is the responsibility of the receiving chain to authenticate that the message was received from owner of address. The receiving chain must store the mapping from: address -> counterpartyAddress for the given channel. Then, onRecvPacket of the destination fee middleware can query for the counterparty address of the recvPacket message sender in order to get the source address of the forward relayer. This source address is what will get embedded in the acknowledgement.
```

I'd use spec terminology of address -> counterpartyAddress, but the godoc should specify this is the address on the source chain of the forward relayer and that this is used in processing a packet recv on the destination chain

Reply:
Updated. wdyt?